modules/s3db/inv.py

```diff
@@ -2142,9 +2141,11 @@ def prep(r):
                     # "received" must not propagate:
                     del request.get_vars["received"]
                     # Set the items to being received
-                    # @ToDo: Check Permissions & Avoid DB updates in GETs
-                    db(sendtable.id == r.id).update(status = SHIP_STATUS_RECEIVED)
-                    db(tracktable.send_id == r.id).update(status = TRACK_STATUS_ARRIVED)
+                    if request.env.request_method != 'GET':
```

Comment:
r.http, Ideally you should use the S3Request object(atleast for s3_rest_controllers) to deal with the request-related info.